src/vm/class.cpp

```diff
@@ -603,12 +603,17 @@ HRESULT EEClass::AddMethod(MethodTable * pMT, mdMethodDef methodDef, RVA newRVA,
     // Get the new MethodDesc (Note: The method desc memory is zero initialized)
     MethodDesc *pNewMD = pChunk->GetFirstMethodDesc();
 
-    ACQUIRE_STACKING_ALLOCATOR(pStackingAllocator);
 
     // Initialize the new MethodDesc
+    // The stacking allocator being passed here as NULL replaces a silent optimization bug that was being masked before here.
+    // This was reported in https://github.com/dotnet/coreclr/issues/26016, and was caused by PR https://github.com/dotnet/coreclr/pull/24177
+    // Previously, &GetThread()->m_MarshalAlloc was being optimized into an offset from a globl thread-static. After the PR, this resulted
```

Comment:
The comment that describes the history is appropriate for PR descriptions, but not really helpful in the code.